DEVELOPMENT.md

```
@@ -0,0 +1,306 @@
+# Run type check in watch mode (continuous)
+npm run typecheck:watch
+
+# Type check with pretty output
+npm run lint:types
+
+# Legacy type check using standard TypeScript (slower)
+npm run typecheck:legacy
+```
+
+**Performance:** The project uses `@typescript/native-preview` (TypeScript Native Port) which is significantly faster than the standard TypeScript compiler. This is especially noticeable on larger codebases or in watch mode.
+
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)